### DISCUSSION TITLE
IOT simplified version for sensors and the like.

### DISCUSSION THREAD

**Discussion by Yardie-** [oetiker/rrdtool-1.x#1213] [Ideas]
Hi 
I used MRTG and RRDtool many years ago and really love it for network monitoring. 
I am wondering if anyone has done a simplified stripped down version for the modern SOC chips like the ESP32.
It seems to me that the RRDtool database would be perfect for long term sensor data storage.
The many graphics libraries available for TFT screens and the like could easily be used to display data live. But a simple data access API would suffice to allow users to do their own data visualisation.
I realise RRDtool is now extremely complex covering many many systems so this may be a rather large undertaking.

**Comment by oetiker** [comment]
Hi Yardie, I am not aware of such a project ... but I guess it should be possible ... I would certainly accept PRs for patches to integrated the port! Sounds like an exciting project :)

**Reply by Yardie-** [reply]
Ok I will have a bang at it later in the year. 
Not sure if a PR would work as it would be a major strip down of the library.
Basically taking it back to just a database and an insertion retrieval system with a basic API.

**Reply by oetiker** [reply]
the components are quite independent, so it would probably suffice to just not compile the graphics things :)

**Reply by Yardie-** [reply]
Ok well that should be heaps easier then.
I'm going away for a bit soon so I'll take the code and get my head around it.
